src/python/pants/backend/python/dependency_inference/module_mapper.py

```diff
@@ -441,12 +446,29 @@ async def map_module_to_address(
         if possible_provider.ancestry == val[0]:
             val[1].append(possible_provider.provider)
 
-    closest_providers: list[ModuleProvider] = list(
+    if request.locality:
+        # For each provider type, if we have more than one provider left, prefer
+        # the one with the closest common ancestor to the requester.
```

Comment:
This doesn't actually check the "if we have more than one provider left" condition, but probably should to avoid the `for provider in providers` loop entirely?